optimum/intel/openvino/modeling_diffusion.py

```diff
@@ -0,0 +1,229 @@
+#  Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+import os
+from pathlib import Path
+from typing import Optional, Union
+
+import numpy as np
+
+import openvino
+from huggingface_hub import HfApi, hf_hub_download
+from openvino.offline_transformations import compress_model_transformation
+from openvino.runtime import Core
+from optimum.modeling_base import OptimizedModel
+
+from .utils import ONNX_WEIGHTS_NAME, OV_XML_FILE_NAME
+
+
+core = Core()
+
+logger = logging.getLogger(__name__)
+
+
+class OVModelForStableDiffusion:
+    def __init__(self, model: openvino.runtime.Model, **kwargs):
+        logger.info(
+            "`optimum.intel.openvino.OVModelForStableDiffusion` is experimental and might change in the future."
+        )
+        self.model = model
+        self._device = kwargs.get("device", "CPU")
+        self.ov_config = {"PERFORMANCE_HINT": "LATENCY"}
+        self.request = None
+
+        # Reshape the U-NET model
+        if "encoder_hidden_states" in [inputs.get_any_name() for inputs in self.model.inputs]:
+            self.model = self._reshape(self.model)
```

Comment:
Would it help to have this operation in diffusers during export?

Reply:
Yes I'll check if by modifying the ONNX export, we can make this reshaping step optional